plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/AppEngineProjectIdValidator.java

```diff
@@ -0,0 +1,29 @@
+package com.google.cloud.tools.eclipse.appengine.newproject;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
```

Comment:
This is the project ID shown on the developer console. It has basically two forms:

word-word
domain.com:word-word

The third form shown in the Go binding tests that adds a partition is an internal form not exposed externally.